services/facade/src/put.js

```diff
@@ -0,0 +1,45 @@
+import AWS from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies
+
+const region = process.env.REGION;
+
+const s3 = new AWS.S3({ signatureVersion: 'v4', region });
+
+/* eslint-disable import/prefer-default-export, no-console */
+export const handler = (event, context, callback) => {
+  const file = event.headers['x-amz-meta-producer-key'];
+
+  if (!file) {
+    const response = {
+      statusCode: 400,
+      body: JSON.stringify({
+        message: `Missing x-amz-meta-producer-key header`,
+      }),
+    };
+
+    callback(null, response);
+  }
+
+  // If producer has correctly submitted a key.
+  const params = {
+    Bucket: 'eubfr-dev',
+    Key: file,
+    Expires: 30,
+  };
+
+  s3.getSignedUrl('putObject', params, (err, url) => {
+    if (err) {
+      callback(err, 'Error creating a signed upload.');
```

Comment:
The second parameter of the callback is for successful operations — https://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-handler.html#nodejs-prog-model-handler-callback